src/qibo/backends/hamming_weight.py

```diff
@@ -1,716 +1,80 @@
-    ):
-        """Dirac notation representation of the state in the computational basis.
-
-        Args:
-            state (ndarray): state to suffer the action of ``gate``.
-            nqubits (int): total number of qubits in the circuit.
-            weight (int): Hamming-weight of ``state``.
-            decimals (int, optional): Number of decimals for the amplitudes.
-                Defaults to :math:`5`.
-            cutoff (float, optional): Amplitudes with absolute value smaller than the
-                cutoff are ignored from the representation. Defaults to  :math:`1e-10`.
-            max_terms (int, optional): Maximum number of terms to print. If the state
-                contains more terms they will be ignored. Defaults to :math:`20`.
-
-        Returns:
-            str: String representing the state in the computational basis.
-        """
-        state = self.to_numpy(state)
-        terms = []
-
-        if (
-            self._dict_indexes is None
-            or list(self._dict_indexes.keys())[0].count("1") != weight
-        ):
-            self._dict_indexes = self._get_lexicographical_order(nqubits, weight)
-
-        strings = list(self._dict_indexes.keys())
-        for i in self.np.nonzero(state)[0]:
-            i = int(i)
-            b = strings[i]
-            if self.np.abs(state[i]) >= cutoff:
-                x = self.np.round(state[i], decimals)
-                terms.append(f"{x}|{b}>")
-            if len(terms) >= max_terms:
-                terms.append("...")
-                return terms
-        return terms
-
-    def calculate_probabilities(self, state, qubits, weight, nqubits):
-        """Calculate the probabilities of the measured qubits from the statevector.
-
-        Args:
-            state (ndarray): state to suffer the action of ``gate``.
-            qubits (list or set, optional): Set of qubits that are measured.
-            weight (int): Hamming-weight of ``state``.
-            nqubits (int): total number of qubits in the circuit.
-
-        Returns:
-            ndarray: Probabilities over the input qubits.
-        """
-        rtype = self.np.real(state).dtype
-
-        if (
-            self._dict_indexes is None
-            or list(self._dict_indexes.keys())[0].count("1") != weight
-        ):
-            self._dict_indexes = self._get_lexicographical_order(nqubits, weight)
-
-        strings = list(self._dict_indexes.keys())
-        indexes = [index[0] for index in self._dict_indexes.values()]
-
-        measured_strings = {}
-        for string, index in zip(strings, indexes):
-            measured_string = "".join(string[q] for q in qubits)
-            measured_strings[measured_string] = measured_strings.get(
-                measured_string, 0
-            ) + self.np.real(self.np.abs(state[index]) ** 2)
+    spec = find_spec("qibo.backends._hamming_weight_operations")
+    module = module_from_spec(spec)
+    spec.loader.exec_module(module)
```

Comment:
can't you just
```suggestion
    import qibo.backends._hamming_weight_operations as module
```
?
since you always import the methods from the same file and it's not runtime dependent